src/views/GraphTest.vue

```diff
@@ -161,93 +176,35 @@
-                        }
-                    });
-                });
-                let newRelations = await Promise.all(newIDs.map(id => {
-                    if (id) {
-                        let subQuery = JSON.parse(JSON.stringify(graphQuery));
-                        subQuery.queryParam.id = id;
-                        return graphClient.executeQuery(subQuery);
-                    }
-                }));
-                newRelations.forEach(newRelation => {
-                  if (Object.keys(newRelation).includes('fairsharingRecord')) {
-                    let subAssociations = newRelation.fairsharingRecord.recordAssociations;
-                    subAssociations.forEach(subAssociation => {
-                      if (newIDs.includes(subAssociation.linkedRecord.id)) {
-                        seriesData.push([
-                          newRelation.fairsharingRecord.metadata.name,
-                          subAssociation.linkedRecord.name,
-                          subAssociation.recordAssocLabel.replace(/_/g, " ").toUpperCase()
-                        ]);
-                        /*nodes.push({
-                                id: subAssociation.linkedRecord.name,
-                                color: 'orange',
-                                marker: {
-                                    radius: 5
-                                }
-                            });*/
-                      }
-                    });
-                  }
-                });
-                if (nodes.length > 200){
+
+                this.options.series[0].data = response.fairsharingGraph.edges;
+                this.options.series[0].nodes = response.fairsharingGraph.nodes;
+                this.options.subtitle.text = this.options.series[0].nodes[0].id + ' Network Graph';
+
+                if (this.options.series[0].nodes.length > 200){
                   this.options.plotOptions.networkgraph.layoutAlgorithm.linkLength = 40;
                   this.options.plotOptions.networkgraph.layoutAlgorithm.maxIterations = 50;
                 }
-                else if (nodes.length > 100){
+                else if (this.options.series[0].nodes.length > 100){
                   this.options.plotOptions.networkgraph.layoutAlgorithm.linkLength = 60;
                   this.options.plotOptions.networkgraph.layoutAlgorithm.maxIterations = 100;
                 }
-                else if (nodes.length < 30) {
+                else if (this.options.series[0].nodes.length < 30) {
                   this.options.plotOptions.networkgraph.layoutAlgorithm.linkLength = 80;
                   this.options.plotOptions.networkgraph.layoutAlgorithm.maxIterations = 300;
                 }
                 else {
                   this.options.plotOptions.networkgraph.layoutAlgorithm.linkLength = 60;
                   this.options.plotOptions.networkgraph.layoutAlgorithm.maxIterations = 120;
                 }
```

Comment:
Theere's another bug, see https://deploy-preview-830--fairsharing.netlify.app/#/graph/1414 where the graph doesn't load at all. We get fairsharingGraph is undefined. Probably because there's no relationships.

Reply:
In a roundabout way, yes. The graph will display if there are no relationships (just node 0). But, something else trivial to fix on the server was broken in that circumstance. I am pushing the fix now.